test/server/cards/06-CotE/IsawaUjina.spec.js

```diff
@@ -0,0 +1,133 @@
+                    expect(this.player1).toBeAbleToSelect(this.isawaUjina);
+                    expect(this.player1).toBeAbleToSelect(this.callowDelegate);
+                    expect(this.player1).toBeAbleToSelect(this.dojiChallenger);
+                    expect(this.player1).not.toBeAbleToSelect(this.dojiHotaru);
+                });
+
+                it('should remove the target from the game', function() {
+                    this.player1.clickCard(this.dojiChallenger);
+                    expect(this.dojiChallenger.location).toBe('removed from game');
+                });
+
+                it('should trigger \'leaves play\' triggers', function() {
+                    this.player1.clickCard(this.callowDelegate);
+                    expect(this.player2).toHavePrompt('Triggered Abilities');
+                    expect(this.player2).toBeAbleToSelect(this.callowDelegate);
+                });
+            });
+
+            it('should trigger multiple times', function() {
+                this.noMoreActions();
+                this.initiateConflict({
+                    attackers: [this.isawaUjina],
+                    defenders: [this.dojiChallenger],
+                    type: 'political',
+                    ring: 'void'
+                });
+                this.noMoreActions();
+                this.player1.clickCard(this.dojiChallenger);
+                expect(this.player1).toHavePrompt('Isawa Ujina');
+                this.player1.clickCard(this.dojiChallenger);
+                expect(this.dojiChallenger.location).toBe('removed from game');
+                this.player1.clickCard(this.knowTheWorld1);
+                this.player1.clickRing('void');
+                this.player1.clickRing('air');
+                this.noMoreActions();
+                this.initiateConflict({
+                    attackers: [this.callowDelegate],
+                    defenders: [this.adeptOfTheWaves],
+                    type: 'political',
+                    ring: 'void'
+                });
+                this.noMoreActions();
+                expect(this.player1).toHavePrompt('Isawa Ujina');
+                this.player1.clickCard(this.callowDelegate);
+                this.player2.clickPrompt('Pass');
+                expect(this.callowDelegate.location).toBe('removed from game');
+                this.player1.clickCard(this.knowTheWorld2);
+                this.player1.clickRing('void');
+                this.player1.clickRing('earth');
+                this.noMoreActions();
+                this.initiateConflict({
+                    attackers: [this.fushicho],
+                    defenders: [],
+                    ring: 'void'
+                });
+                this.noMoreActions();
+                this.player1.clickPrompt('No');
+                this.player1.clickCard(this.dojiHotaru);
+                expect(this.player1).toHavePrompt('Isawa Ujina');
+            });
```

Comment:
Might be worth a test making sure it interacts correctly with Kaede?

Reply:
Done.